pandas/indexes/numeric.py

```diff
@@ -22,6 +22,36 @@ class NumericIndex(Index):
     """
     _is_numeric_dtype = True
 
+    def __new__(cls, data=None, dtype=None, copy=False, name=None,
+                fastpath=False):
+
+        if fastpath:
+            return cls._simple_new(data, name=name)
+
+        if dtype and np.dtype(dtype) != cls._default_dtype:
```

Comment:
the dtype argument is only a suggestion. We don't support actual ``int32`` dtypes withing Int64Index for example, its an implementation detail. It actually is sometimes conveinent to be able to put a dtype specification here (to make this more general); in reaility ``dtype`` is NOT a public kwarg.

that's how it is, pls don't change it.